kolibri/core/tasks/api.py

```diff
@@ -197,16 +198,25 @@ def initial(self, request, *args, **kwargs):
             self.permission_classes = []
         return super(BaseViewSet, self).initial(request, *args, **kwargs)
 
-    def default_permission_classes(self):
-        # For all /api/tasks/ endpoints
-        return [CanManageContent]
+    def check_registered_job_permissions(self, request, registered_job):
+        """
+        Checks whether the `request` is allowed to proceed based on
+        `registered_job.permissions`.
+
+        Raises `PermissionDenied` if `request.user` is not allowed to proceed.
+        """
+        for permission in registered_job.permissions:
+            if not permission.has_permission(request, self):
```

Comment:
One slight issue with just using the 'registered_job' and not the job object itself is that we might need more granular control.

For example, for the 'facility' queue jobs, we only want to allow a user to see jobs for their own facility (if they have appropriate permissions) and not for all facilities, to do this, we would have to introspect the Job object.

The superuser is of course the exception, as they can see all facility jobs across all facilities.

Reply:
@rtibbles do we have a DRF permission on the lines of `CanManageContentOfOwnFacility` or can we build one like we have here: https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/core/tasks/permissions.py...?

Reply:
My thought was to do something similar to the `readable_by_user_filter` that we have on the Kolibri permissions classes: https://github.com/learningequality/kolibri/blob/b87f9da274a6a9218419cf9f858be695aa04304a/kolibri/core/auth/permissions/base.py#L57

We couldn't use it directly though, as it would have to be compatible with the SQLAlchemy interface we are using for querying jobs.

Reply:
Or... because it depends on metadata inside the stored job object, it would probably have to do the filtering in Python anyway.

Reply:
Resolved on Slack.